main/src/main/scala/sbt/Main.scala

```diff
@@ -943,4 +946,36 @@ object BuiltinCommands {
     Command.command(NotifyUsersAboutShell) { state =>
       notifyUsersAboutShell(state); state
     }
+
+  private[this] def skipWelcomeFile(state: State, version: String) = {
+    val base = BuildPaths.getGlobalBase(state).toPath
+    base.resolve("preferences").resolve(version).resolve(SkipWelcomeFileName)
+  }
+  private def displayWelcomeBanner(state: State): State = {
+    if (!state.get(welcomeHasBeenShown).getOrElse(false)) {
+      try {
+        val version = sbtVersion(state)
+        val skipFile = skipWelcomeFile(state, version)
+        Files.createDirectories(skipFile.getParent)
+        val suppress = !SysProp.banner || Files.exists(skipFile)
+        if (!suppress) state.log.info(WelcomeBanner(version))
+      } catch { case _: IOException => /* Don't let errors in this command prevent startup */ }
+      state.put(welcomeHasBeenShown, true)
+    } else state
+  }
+  private[this] val welcomeHasBeenShown =
+    AttributeKey[Boolean]("welcome-banner-has-been-shown", Int.MaxValue)
+  private[this] val SkipWelcomeFileName = "skip-welcome"
+  private[this] val SkipWelcomeBanner = "skipWelcomeBanner"
```

Comment:
There's something awkward about telling someone to skip the welcoming, so it might be better to call it "skipBanner"?

Reply:
Ok. https://github.com/sbt/sbt/compare/f0f5eb064a9ce479d13152afa00e07bc8f5bc8c0..5d2c60496f977c87b6f02b523df1052ae6b722f7